core/dblookupext/historyRepository.go

```diff
@@ -246,12 +247,14 @@ func (hr *historyRepository) OnNotarizedBlocks(shardID uint32, headers []data.He
 	for i, headerHandler := range headers {
 		headerHash := headersHashes[i]
 
+		log.Debug("onNotarizedBlocks():", "shardID", shardID, "nonce", headerHandler.GetNonce(), "headerHash", headerHash, "type", fmt.Sprintf("%T", headerHandler))
+
+		// Question for review: this optimization is not required anymore, right?
```

Comment:
The only situation when the notifications will be redundant will be when a node will do a rollback. In this situation the same final block/blocks will be notified again, after another new block will be constructed upon it. But this case should not be a regular pattern and could be considered as an exception. If this redundant notifications will not break something in the implementation logic and the below method was used just as an optimization, I would say that you can remove it now, considering the new approach.